src/handlers/on_ticket.py

```diff
@@ -127,8 +127,12 @@ def fetch_file_contents_with_retry():
     #     repo_full_name,
     #     installation_id=installation_id,
     # )
-
-    logger.info("Fetching relevant files...")
+    if current_issue.state == 'closed':
+        logger.info('Issue is closed, not retrying.')
+logger.info('Issue is closed, not retrying.')
```

Comment:
deduplicate these and move the posthog capture under the if statement. Make sure to log the username